DependencyInjection/DoctrineExtension.php

```diff
@@ -927,4 +936,25 @@ private function createArrayAdapterCachePool(ContainerBuilder $container, string
 
         return $id;
     }
+
+    private function registerMetadataPhpArrayCaching(string $entityManagerName, ContainerBuilder $container): void
+    {
+        $metadataCacheAlias              = $this->getObjectManagerElementName($entityManagerName . '_metadata_cache');
+        $decoratedMetadataCacheServiceId = (string) $container->getAlias($metadataCacheAlias);
+        $phpArrayCacheDecoratorServiceId = $decoratedMetadataCacheServiceId . '.php_array';
```

Comment:
Why is there decoration involved?

Reply:
PhpArrayAdapter requires it https://github.com/symfony/symfony/blob/57e39b41b94dd9d0beb04c7f88fc0955a5bbe605/src/Symfony/Component/Cache/Adapter/PhpArrayAdapter.php#L48. Also for BC, so current configuration is not completely ignored until we remove it.